packages/plugin-react/src/index.ts

```diff
@@ -240,18 +249,6 @@ export default function viteReact(opts: Options = {}): PluginOption[] {
                 : [null, false]
 
             if (isJSX || (ast = restoredAst)) {
-              plugins.push([
```

Comment:
Because we're not adding any plugins anymore, the `shouldSkip` test at line 290 will always return true for external dependencies except when the dep is in a monorepo so the `isNodeModules` test returns false causing `isProjectfile` to return true. Under any other circumstance, the `restoreJsx` transform is applied but the result is thrown away.

There is another issue with `restoreJsx`, it doesn't work on optimized deps because its the `parseReactAlias` function isn't smart enough to detect `React.createElement` calls in optimized deps.

For these two reasons, the `restoreJsx` transform doesn't fulfill its raison d'être (transforming deps to make them use the automatic runtime). And under the rare circumstances that it does work (the dep is in a monorepo and it's excluded from `optmizeDeps`) this PR will cause it to send JSX to the browser, causing syntax errors (because it's applied _after_ the ESBuild JSX transform).

Reply:
I hope this comment makes the issue clear, it's very hard (and kind of pointless) to provide a repro for this behavior.

Reply:
To me, it seems `restoreJsx` never worked as intended anyway. Maybe we could just remove it?

Reply:
I have no objections to either and can happily contribute 2 if it is acceptable for it to be a Babel plugin itself. I don't know what 1 would entail though,  it touches parts of the code base I'm not very familiar with.

Reply:
Actually it _is_ a Babel plugin, we just need to improve `parseReactAlias`.